tikv/pd
# scripts: add the inefficient assert function usage check

Signed-off-by: JmPotato <ghzpotato@gmail.com>



### What problem does this PR solve?


Issue Number: Close #5104. Should be merged after https://github.com/tikv/pd/pull/5109.

### What is changed and how does it work?



```commit-message
Add the inefficient assert function usage check.
```

### Check List



Tests



- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)

```shell
./scripts/check-test.sh
following packages use inefficient assert function: please replace reflect.DeepEqual with require.Equal
./client/client_test.go:61:	re.True(reflect.DeepEqual(getURLs([]*pdpb.Member{members[1], members[3], members[2]}), cli.GetURLs()))
./client/client_test.go:63:	re.True(reflect.DeepEqual(getURLs([]*pdpb.Member{members[1], members[3], members[2]}), cli.GetURLs()))
./client/client_test.go:65:	re.True(reflect.DeepEqual(getURLs([]*pdpb.Member{members[1], members[3], members[2], members[0]}), cli.GetURLs()))
./pkg/autoscaling/calculation_test.go:200:			re.True(reflect.DeepEqual(testCase.expectedPlan, plans))
./pkg/cache/cache_test.go:109:	re.True(reflect.DeepEqual(sortIDs(cache.GetAllID()), []uint64{3}))
./pkg/cache/cache_test.go:128:	re.True(reflect.DeepEqual(val, "3"))
./pkg/cache/cache_test.go:132:	re.True(reflect.DeepEqual(val, "2"))
./pkg/cache/cache_test.go:136:	re.True(reflect.DeepEqual(val, "1"))
./pkg/cache/cache_test.go:150:	re.True(reflect.DeepEqual(val, "1"))
./pkg/cache/cache_test.go:154:	re.True(reflect.DeepEqual(val, "2"))
./pkg/cache/cache_test.go:158:	re.True(reflect.DeepEqual(val, "4"))
./pkg/cache/cache_test.go:164:	re.True(reflect.DeepEqual(val, "1"))
./pkg/cache/cache_test.go:168:	re.True(reflect.DeepEqual(elems[0].Value, "4"))
./pkg/cache/cache_test.go:169:	re.True(reflect.DeepEqual(elems[1].Value, "2"))
./pkg/cache/cache_test.go:170:	re.True(reflect.DeepEqual(elems[2].Value, "1"))
./pkg/cache/cache_test.go:208:	re.True(reflect.DeepEqual(elems[0].Value, "2"))
./pkg/cache/cache_test.go:209:	re.True(reflect.DeepEqual(elems[1].Value, "3"))
./pkg/cache/cache_test.go:210:	re.True(reflect.DeepEqual(elems[2].Value, "4"))
./pkg/cache/cache_test.go:214:	re.True(reflect.DeepEqual(elems[0].Value, "4"))
./pkg/cache/cache_test.go:231:	re.True(reflect.DeepEqual(val, "3"))
./pkg/cache/cache_test.go:235:	re.True(reflect.DeepEqual(val, "2"))
./pkg/cache/cache_test.go:239:	re.True(reflect.DeepEqual(val, "1"))
./pkg/cache/cache_test.go:253:	re.True(reflect.DeepEqual(val, "1"))
./pkg/cache/cache_test.go:257:	re.True(reflect.DeepEqual(val, "2"))
./pkg/cache/cache_test.go:261:	re.True(reflect.DeepEqual(val, "4"))
./pkg/cache/cache_test.go:267:	re.True(reflect.DeepEqual(val, "1"))
./pkg/cache/cache_test.go:271:	re.True(reflect.DeepEqual(elems[0].Value, "4"))
./pkg/cache/cache_test.go:272:	re.True(reflect.DeepEqual(elems[1].Value, "2"))
./pkg/cache/cache_test.go:273:	re.True(reflect.DeepEqual(elems[2].Value, "1"))
./pkg/cache/cache_test.go:79:	re.True(reflect.DeepEqual(sortIDs(cache.GetAllID()), []uint64{1, 2, 3}))
./pkg/cache/cache_test.go:96:	re.True(reflect.DeepEqual(sortIDs(cache.GetAllID()), []uint64{2, 3}))
./pkg/logutil/log_test.go:76:			re.True(reflect.DeepEqual(testCase.expect, RedactBytes(r)))
./pkg/logutil/log_test.go:78:			re.True(reflect.DeepEqual(testCase.expect, RedactString(r)))
./pkg/logutil/log_test.go:80:			re.True(reflect.DeepEqual(testCase.expect, RedactStringer(r)))
./pkg/typeutil/conversion_test.go:38:	re.True(reflect.DeepEqual([]byte(str), b))
./pkg/typeutil/string_slice_test.go:35:	re.True(reflect.DeepEqual(b, nb))
./pkg/typeutil/string_slice_test.go:47:	re.True(reflect.DeepEqual(ss, ss2))
./tests/client/client_test.go:1039:	suite.True(reflect.DeepEqual(store, n))
./tests/client/client_test.go:1056:	suite.True(reflect.DeepEqual(offlineStore, n))
./tests/client/client_test.go:1065:			suite.True(reflect.DeepEqual(offlineStore, store))
./tests/client/client_test.go:116:	re.True(reflect.DeepEqual(endpoints, urls))
./tests/client/client_test.go:267:			re.True(reflect.DeepEqual(endpoints, urls))
./tests/client/client_test.go:977:			suite.True(reflect.DeepEqual(expect[i], scanRegions[i].Meta))
./tests/client/client_test.go:980:				suite.True(reflect.DeepEqual(&metapb.Peer{}, scanRegions[i].Leader))
./tests/client/client_test.go:982:				suite.True(reflect.DeepEqual(expect[i].Peers[0], scanRegions[i].Leader))
./tests/client/client_test.go:986:				suite.True(reflect.DeepEqual([]*metapb.Peer{expect[i].Peers[1]}, scanRegions[i].DownPeers))
./tests/client/client_test.go:990:				suite.True(reflect.DeepEqual([]*metapb.Peer{expect[i].Peers[1], expect[i].Peers[2]}, scanRegions[i].PendingPeers))
make: *** [check-test] Error 1
```

### Release note



```release-note
None.
```
